server/app/services/scheduler/filter/volume_plugin.rb

```diff
@@ -19,7 +19,8 @@ def for_service(service, instance_number, nodes)
         }.compact
 
         nodes = nodes.select { |node|
-          (needed_drivers - node.plugins['volume']).empty?
+          volume_plugins = node.plugins['volume'].map { |v| v['name'] }
```

Comment:
AFAIK if the plugin name is something like `rexray/s3fs:latest`, then the volume driver name is just `rexray/s3fs`? Or was it even just `s3fs`?

This would read a lot better if it was

```ruby
volume_drivers = node.volume_plugins.map { |v| v['name'].split(':')[0] }
```

Notice how the "plugins" turn into "drivers"... maybe the agent could parse out the `name` and `version` fields separately?